src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,98 @@
+# we consider the ODE dxₜ/dt = f(xₜ,λ(rt))
+# here λ = λ(t) ∈ Rᵐ is a function containing all the system parameters 
+
+# We ask the user to define: 
+#  1) a ContinuousTimeDynamicalSystem that should be investigated and
+#  2) a protocol for the time-dependent forcing with the struct RateProtocol
+
+# Then we give back the ContinuousTimeDynamicalSystem with the parameter 
+# changing according to the rate protocol
+"""
+    RateProtocol
+
+The RateProtocol contains all the fields (information) that allows to take an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ)` 
+with `λ ∈ Rᵐ` containing all system parameters, and make it an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ(rt))`
+with `λ(t) ∈ Rᵐ` constant before time `t_start` and also after time `t_end`. 
+
+RateProtocol contains the following fields:
+- `λ::Function`: forcing function of the form `λ(p, t_start; kwargs...)``
+- `p_lambda::Vector`: parameters of forcing function
+- `r::Float64`: rate parameter
+- `t_start::Float64`: start time of protocol
+- `t_end::Float64`: end time of protocol
+
+Default values 
+==============
+
+If `t_start` and `t_end` are not provided, they are set to `t_start=-Inf`, and `t_end=Inf`.
+If `p_lambda` is not provided, it is set to an empty array `[]`.
```

Comment:
You can avoid all of this by simply assigning values to when you stated the keywords. E.g., `t_end = Inf`.